File: android/designsystem/src/main/kotlin/com/cacompte/designsystem/previews/IdentityGallery.kt
package com.cacompte.designsystem.previews

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.cacompte.designsystem.components.Avatar
import com.cacompte.designsystem.components.AvatarKind
import com.cacompte.designsystem.components.AvatarSize
import com.cacompte.designsystem.components.AvatarView
import com.cacompte.designsystem.components.PlayerPalette
import com.cacompte.designsystem.components.color
import com.cacompte.designsystem.theme.CaCompteTheme

/**
 * Galerie des 10 couleurs joueur et des 3 tailles d'`AvatarView` — équivalent de
 * `IdentityGallery.swift` (sans le logo, qui n'existe pas encore côté Android à cette étape).
 * Variantes clair/sombre uniquement, pas d'AX5 — même choix que la source Swift.
 */
@Composable
private fun IdentityGalleryContent() {
    Surface {
        Column(
            modifier = Modifier.fillMaxWidth().padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
        ) {
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                for (index in 1..10) {
                    val palette = PlayerPalette(index)
                    Box(
                        modifier =
                            Modifier
                                .size(28.dp)
                                .clip(CircleShape)
                                .background(palette.color()),
                    )
                }
            }

            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                AvatarView(avatar = Avatar.generated("Alice"), size = AvatarSize.Small)
                AvatarView(avatar = Avatar.generated("Bob"), size = AvatarSize.Medium)
                AvatarView(avatar = Avatar.generated("Chloé"), size = AvatarSize.Large)
                AvatarView(
                    avatar = Avatar(kind = AvatarKind.Emoji("🎲"), palette = PlayerPalette(3)),
                    size = AvatarSize.Medium,
                )
            }
        }
    }
}

@Preview(name = "Identité — clair", showBackground = true)
@Composable
private fun IdentityGalleryLightPreview() {
    CaCompteTheme(darkTheme = false) { IdentityGalleryContent() }
}

@Preview(name = "Identité — sombre", showBackground = true)
@Composable
private fun IdentityGalleryDarkPreview() {
    CaCompteTheme(darkTheme = true) { IdentityGalleryContent() }
}

File: apple/CaCompteKit/Sources/DesignSystem/Previews/IdentityGallery.swift
import SwiftUI

#Preview("Joueurs & logo — clair") {
  IdentityGallery()
}

#Preview("Joueurs & logo — sombre") {
  IdentityGallery()
    .preferredColorScheme(.dark)
}

private struct IdentityGallery: View {
  var body: some View {
    ScrollView {
      VStack(alignment: .leading, spacing: Space.xl) {
        VStack(alignment: .leading, spacing: Space.sm) {
          Text("Palette joueurs").font(.h6).foregroundStyle(.textSecondary)
          HStack(spacing: Space.sm) {
            ForEach(1...10, id: \.self) { index in
              Circle()
                .fill(PlayerPalette(index: index).color)
                .frame(width: 28, height: 28)
            }
          }
        }

        VStack(alignment: .leading, spacing: Space.sm) {
          Text("Avatars").font(.h6).foregroundStyle(.textSecondary)
          HStack(spacing: Space.md) {
            AvatarView(avatar: .generated(for: "Alice"), size: .small)
            AvatarView(avatar: .generated(for: "Bob"), size: .medium)
            AvatarView(avatar: .generated(for: "Chloé"), size: .large)
            AvatarView(
              avatar: Avatar(kind: .emoji("🦊"), palette: PlayerPalette(index: 2)), size: .medium)
          }
        }

        VStack(alignment: .leading, spacing: Space.sm) {
          Text("Logo").font(.h6).foregroundStyle(.textSecondary)
          LogoMarkView()
            .frame(width: 96, height: 96)
            .background(.neutralSurface, in: .rect(cornerRadius: Radius.md))
          LogoLockupHorizontal(height: 40)
            .background(.neutralSurface, in: .rect(cornerRadius: Radius.md))
          LogoLockupVertical(height: 40)
            .background(.neutralSurface, in: .rect(cornerRadius: Radius.md))
        }
      }
      .padding(Space.lg)
    }
    .background(.neutralBg)
  }
}
